search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/IndexFilterTransformer.java

```diff
@@ -18,6 +34,80 @@ public IndexFilterTransformer(final IndexDescriptor indexDescriptor) {
     this.indexDescriptor = indexDescriptor;
   }
 
+  public SearchQuery toSearchQuery(
+      final T filter, final ResourceAccessChecks resourceAccessChecks) {
+    final var filterSearchQuery =
+        Optional.ofNullable(toSearchQuery(filter)).orElseGet(SearchQueryBuilders::matchAll);
+
+    if (resourceAccessChecks == null) {
+      return rewriteSearchQueries(List.of(filterSearchQuery));
+    }
+
+    final var authorizationSearchQuery =
+        Optional.of(resourceAccessChecks)
+            .map(ResourceAccessChecks::authorizationCheck)
+            .map(this::applyAuthorizationChecks)
+            .orElse(matchAll());
```

Comment:
💭 Not a big deal but given that we already check for null above, we can simplify this as below. Same for the tenant search:

```
final var authorizationCheck = resourceAccessChecks.authorizationCheck();
final var authorizationSearchQuery = applyAuthorizationChecks(authorizationCheck);
```